compiler-rt/lib/profile/InstrProfilingFile.c

```diff
@@ -1259,7 +1259,7 @@ int __llvm_profile_register_write_file_atexit(void) {
   lprofSetupValueProfiler();
 
   HasBeenRegistered = 1;
-  return atexit(writeFileWithoutReturn);
+  return lprofAtExit(writeFileWithoutReturn);
```

Comment:
`fn_stack_push` tries to be thread safe, but this `HasBeenRegistered` assignment is not.

Reply:
True. I left it as is because it's existing code, so I'm not regressing anything. A big chunk of the profile runtime's state is not thread safe (for example the part that's initialized by `__llvm_profile_initialize`). 

The reason the stack was made thread safe is because POSIX `atexit` is thread-safe.
The current uses (2 instances) of `lprofAtExit` do not need to be thread safe AFAICT, but we decided to future-proof it.